sources/packages/backend/apps/queue-consumers/src/processors/assessment/_tests_/cancel-application-assessment.processor.e2e-spec.ts

```diff
@@ -0,0 +1,191 @@
+import { ZBClient } from "zeebe-node";
+import { Job } from "bull";
+import { createMock } from "@golevelup/ts-jest";
+import { INestApplication } from "@nestjs/common";
+import { CancelAssessmentQueueInDTO } from "@sims/services/queue";
+import { QueueNames } from "@sims/utilities";
+import {
+  createTestingAppModule,
+  describeProcessorRootTest,
+} from "../../../../test/helpers";
+import { CancelApplicationAssessmentProcessor } from "../cancel-application-assessment.processor";
+import { DataSource, Repository } from "typeorm";
+import {
+  createFakeDisbursementOveraward,
+  saveFakeApplicationDisbursements,
+} from "@sims/test-utils";
+import {
+  ApplicationStatus,
+  DisbursementOveraward,
+  DisbursementSchedule,
+  DisbursementScheduleStatus,
+  StudentAssessment,
+} from "@sims/sims-db";
+import * as faker from "faker";
+
+describe(
+  describeProcessorRootTest(QueueNames.CancelApplicationAssessment),
+  () => {
+    let app: INestApplication;
+    let processor: CancelApplicationAssessmentProcessor;
+    let zbClientMock: ZBClient;
+    let appDataSource: DataSource;
+    let studentAssessmentRepo: Repository<StudentAssessment>;
+    let disbursementOverawardRepo: Repository<DisbursementOveraward>;
+    let disbursementScheduleRepo: Repository<DisbursementSchedule>;
+
+    beforeAll(async () => {
+      const { nestApplication, dataSource, zbClient } =
+        await createTestingAppModule();
+      app = nestApplication;
+      zbClientMock = zbClient;
+      appDataSource = dataSource;
+      studentAssessmentRepo = dataSource.getRepository(StudentAssessment);
+      disbursementOverawardRepo = dataSource.getRepository(
+        DisbursementOveraward,
+      );
+      disbursementScheduleRepo = dataSource.getRepository(DisbursementSchedule);
+      // Processor under test.
+      processor = app.get(CancelApplicationAssessmentProcessor);
+    });
+
+    beforeEach(() => {
+      jest.resetAllMocks();
+    });
+
+    it("Should cancel the assessment pending disbursements and rollback overawards when the cancelled application has overawards and also one sent and one pending disbursements.", async () => {
+      // Arrange
+      const workflowInstanceId = faker.random.number({
+        min: 1000000000,
+        max: 9999999999,
+      });
+      // Application and disbursements.
+      const application = await saveFakeApplicationDisbursements(
+        appDataSource,
+        null,
+        {
+          applicationStatus: ApplicationStatus.Completed,
+          createSecondDisbursement: true,
+        },
+      );
+      // Adjust assessment.
+      application.applicationStatus = ApplicationStatus.Cancelled;
+      const studentAssessment = application.currentAssessment;
+      studentAssessment.assessmentWorkflowId = workflowInstanceId.toString();
+      await studentAssessmentRepo.save(application.currentAssessment);
```

Comment:
Just to check, is the application status updated through the cascade by saving the assessment?

I was assuming that to the cascade to happen, we muse do like this. `studentAssessment .application = application`

Reply:
Is this because the cascade happens only for update? as per model?

Reply:
Line 72 was added by mistake during a conversation with @ann-aot. I will be removing it.

Reply:
If that is the case, I am confused here. If application status in not either cancelled or overwritten, how come the processor will not throw error as per this logic? 

![image](https://user-images.githubusercontent.com/54600590/231606655-72e8593a-0959-4687-bd5c-4eb7bb86ec05.png)

Reply:
@dheepak-aot application status is `canceled`, that was leftover from our conversation.

Reply:
Code adjusted.